tools/filterdiff.py

```diff
@@ -0,0 +1,154 @@
+# /usr/bin/env python3
+import sys
+import os
+import argparse
+import tempfile
+import json
+import subprocess
+from typing import List, Dict, Optional
+
+
+class FilterImageInfo:
+    """
+    Filter specific fields in the image-info that can't be compared
+    together.
+    """
+
+    def __init__(self, imi: Optional[Dict], to_remove: Optional[List]) -> None:
+        self.imi: Dict = imi if imi else {}
+        self.to_remove = to_remove if to_remove else []
+
+    def _tmpfiles_d(self):
+        """
+        Remove certain files from the tpmfiles_d that have a tendency of having
+        a non fixed content.
+        """
+        tmpfilesd = self.imi.get("tmpfiles.d")
+        if tmpfilesd:
+            for _, v in tmpfilesd.items():
+                for trm in self.to_remove:
+                    if trm in v:
+                        del v[trm]
+
+    def _lvm2(self):
+        """
+        LVM2 partitions have a UUID that is not fixed. Replace the value
+        upon comparison time.
+        """
+        partitions = self.imi.get("partitions")
+        if partitions:
+            for partition in partitions:
+                if partition.get("fstype") == "LVM2_member":
+                    partition["uuid"] = "2022-07-01-fixed-uuid"
+
+    def _iso(self):
+        """
+        For isos, the partition UUID is the date of the build. Replace
+        that with a fixed one for the comparison.
+        """
+        if ("image-format" in self.imi and "type" in self.imi["image-format"] and
+                self.imi["image-format"]["type"] == "raw"):
+            if "partitions" in self.imi:
+                for partition in self.imi["partitions"]:
+                    if "fstype" in partition:
+                        if partition["fstype"] == "iso9660":
+                            partition["uuid"] = "2022-07-01-fixed-uuid"
+
+    def apply(self) -> Dict:
+        self._iso()
+        self._lvm2()
+        self._tmpfiles_d()
+        return self.imi
+
+
+def load_item(path):
+    c = None
+    with open(path, "r") as f:
+        c = json.load(f)
+    # Supports raw image-info or DB entry
+    if "image-info" in c:
+        c = c["image-info"]
+        # Load the unwanted files from the DB
+        if "unwanted_tmpfiles_d" in c:
+            unwanted_tmpfiles_d = c["unwanted_tmpfiles_d"]
+    return c
+
+
+def search_unwanted_tmpfiles_d(path1, path2):
```

Comment:
* The function is named `search` because it will try to find `unwanted_tmpfiles_d` from the input file. The input file can be an DB entry or a JSON file containing only the image_info.
* I also considered using sets, the benefit was not really obvious. Since the conversion from List to Set is not super obvious, I mean the loop is only two lines.

Reply:
> * The function is named `search` because it will try to find `unwanted_tmpfiles_d` from the input file. The input file can be an DB entry or a JSON file containing only the image_info.

The bottom line is that it returns the value assigned to `unwanted_tmpfiles_d` key or an empty list. It does not return all instances of it in the dict or index of it or other similar thing that I (personally) would expect from `search`.

I do not see this as a blocker though, so whatever you end up using, I can live with it. In ideal case, this function should not be needed at all IMO.

> * I also considered using sets, the benefit was not really obvious. I mean the loop is only two lines.

The benefit is that it is less code and lower complexity of operations (https://wiki.python.org/moin/TimeComplexity). The data structure is optimized for what you are trying to achieve.

Reply:
Yeah, I agree, good points.